task2/rest-api-basics/web/src/main/java/com/epam/esm/web/controller/TagController.java

```diff
@@ -0,0 +1,59 @@
+package com.epam.esm.web.controller;
+
+import com.epam.esm.service.TagService;
+import com.epam.esm.service.dto.TagDto;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/tags")
+public class TagController {
+
+    @Autowired
+    private TagService tagService;
+
+    /**
+     * Finds all tags
+     * @return list of tags dto
+     */
+
+    @GetMapping
+    public List<TagDto> findAllTags() {
+        return tagService.findAll();
+    }
+
+    /**
+     * Search for a tag by name
+     * @param id - identifier of the tag
+     * @return tag dto
+     */
+
+    @GetMapping("/{id}")
+    public TagDto findTagById(@PathVariable Long id) {
+        return tagService.findById(id);
+    }
+
+    /**
+     * Saves the tag
+     * @param tagDto - received tag dto
+     * @return found tag dto
+     */
+
+    @PostMapping
+    public TagDto saveTag(@RequestBody TagDto tagDto) {
```

Comment:
I think its better to use "save", "find", ... method name, because its obvious that it will be tag from the name of class.